flink-end-to-end-tests/flink-queryable-state-test/pom.xml

```diff
@@ -33,98 +33,54 @@
 	<packaging>jar</packaging>
 
 	<dependencies>
-		<dependency>
-			<groupId>org.apache.flink</groupId>
-			<artifactId>flink-core</artifactId>
-			<version>${project.version}</version>
-		</dependency>
 		<dependency>
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-streaming-java_${scala.binary.version}</artifactId>
 			<version>${project.version}</version>
+			<scope>provided</scope>
 		</dependency>
 		<dependency>
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-statebackend-rocksdb_${scala.binary.version}</artifactId>
 			<version>${project.version}</version>
+			<scope>provided</scope>
 		</dependency>
 	</dependencies>
 
 	<build>
 		<plugins>
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-jar-plugin</artifactId>
-				<version>2.4</version>
-
+				<artifactId>maven-shade-plugin</artifactId>
 				<executions>
 					<execution>
 						<id>QsStateProducer</id>
 						<phase>package</phase>
 						<goals>
-							<goal>jar</goal>
+							<goal>shade</goal>
 						</goals>
 						<configuration>
-							<classifier>QsStateProducer</classifier>
-							<archive>
-								<manifestEntries>
-									<program-class>
-										org.apache.flink.streaming.tests.queryablestate.QsStateProducer
-									</program-class>
-								</manifestEntries>
-							</archive>
+							<finalName>QsStateProducer</finalName>
+							<transformers>
+								<transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+									<mainClass>org.apache.flink.streaming.tests.queryablestate.QsStateProducer</mainClass>
+								</transformer>
+							</transformers>
 						</configuration>
 					</execution>
-				</executions>
-			</plugin>
-
-			<plugin>
-				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-shade-plugin</artifactId>
-				<executions>
-
 					<execution>
 						<id>QsStateClient</id>
```

Comment:
this jar must be a fat jar containing flink-core and possibly other dependencies as it is not run as a flink application.